network/native_endpointclient_linux.go

```diff
@@ -0,0 +1,438 @@
+package network
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/Azure/azure-container-networking/log"
+	"github.com/Azure/azure-container-networking/netio"
+	"github.com/Azure/azure-container-networking/netlink"
+	"github.com/Azure/azure-container-networking/netns"
+	"github.com/Azure/azure-container-networking/network/networkutils"
+	"github.com/Azure/azure-container-networking/platform"
+
+	vishnetlink "github.com/vishvananda/netlink"
+)
+
+const (
+	azureMac         = "12:34:56:78:9a:bc" // Packets leaving the VM should have this MAC
+	loopbackIf       = "lo"                // The name of the loopback interface
+	numDefaultRoutes = 2                   // VNET NS, when no containers use it, has this many routes
+)
+
+var errorNativeEndpointClient = errors.New("NativeEndpointClient Error")
+
+func newErrorNativeEndpointClient(msg string, errStr string) error {
+	return fmt.Errorf("%w : %s : %s", errorNativeEndpointClient, msg, errStr)
+}
+
+type NativeEndpointClient struct {
+	eth0VethName      string //So like eth0
+	ethXVethName      string //So like eth0.X
+	vnetVethName      string //Peer is containerVethName
+	containerVethName string //Peer is vnetVethName
+
+	vnetMac      net.HardwareAddr
+	containerMac net.HardwareAddr
+	ethXMac      net.HardwareAddr
+
+	vnetNSName           string
+	vnetNSFileDescriptor uintptr
+
+	mode           string
+	vlanID         int
+	netnsClient    netns.NetnsInterface
+	netlink        netlink.NetlinkInterface
+	netioshim      netio.NetIOInterface
+	plClient       platform.ExecClient
+	netUtilsClient networkutils.NetworkUtils
+}
+
+func NewNativeEndpointClient(
+	eth0VethName string,
+	ethXVethName string,
+	vnetVethName string,
+	containerVethName string,
+	vnetNSName string,
+	mode string,
+	vlanid int,
+	nl netlink.NetlinkInterface,
+	plc platform.ExecClient,
```

Comment:
That you need to wrap the method signature tells me there are way too many args. Consider creating a config struct which wraps the args that you need to pass in to build this object.
OR, if there is no "magic" happening in this `New`, remove it entirely? Directly instantiating the struct should always be your default new object creation method, up until creating your object is too complicated to trust the consumer of your type to perform correctly.